lib/active_model/serializer/collection_serializer.rb

```diff
@@ -10,8 +10,14 @@ class CollectionSerializer
       def initialize(resources, options = {})
         @root = options[:root]
```

Comment:
Right now, the only scenario tested is essentially:

```diff
@root = collection_root(options)

# condenses `root || derived_root`
def collection_root(options)
  if root = options[:root]
    root
 else
-   key = serializers.first.try(:json_key) || object.try(:name).try(:underscore)
+   key = serializers.first.try(:json_key) || options[:serializer].try!(:allocate).try!(:json_key).try!(pluralize) || object.try(:name).try(:underscore)
    key.try(:pluralize)
  end
end
```

(or a more thorough impl would be something like)

```ruby
@root = collection_root(options)

# condenses `root || derived_root`
def collection_root(options)
  if root = options[:root]
    root
 else
   key = object.empty? && (serializer_class = options[:serializer]) && (serializer_class.allocate.json_key.try!(:pluralize) || serializer_class._type) # shouldn't need the allocate here
    key ||= (serializer = serializers.first) && serializer.json_key
    key ||= object.any? && object.name.underscore
    key ||= '' # oh no! we should never be here. empty collection and no serializer option
    key.pluralize
  end
end
```
I'm also realizing we're inconsistent with object vs. resources.  We should probably always refer to it as object (which is the serializer api for the thing it encapsulates).